include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,1479 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Enrico Seiler <enrico.seiler@fu-berlin.de>
+ * \brief A constexpr bitset implementation with dynamic size at compile time.
+ */
+
+#pragma once
+
+#include <seqan3/core/bit_manipulation.hpp>
+#include <seqan3/core/concept/cereal.hpp>
+#include <seqan3/range/view/repeat_n.hpp>
+
+namespace seqan3
+{
+
+/*!\brief A constexpr bitset implementation with dynamic size at compile time.
+ * \implements seqan3::ReservableContainer
+ * \implements seqan3::Cerealisable
+ * \ingroup container
+ * \tparam capacity_   The capacity of the dynamic bitset
+ *
+ * This implementation of a bitset can be constructed, accessed and modified at compile time.
+ * It has a fixed capacity but a dynamic size and provides all functionality of a sequence container. Note
+ * that it also models a reservable sequence container but all associated member functions are no-op because the
+ * capacity is fixed.
+ */
+
+template <size_t capacity_ = 58>
+class dynamic_bitset
+{
+private:
+    //!\brief Stores the actual data.
+    uint64_t data_{};
+
+    //!\brief The size in bits of the underlying data structure. 64 for `uint64_t`.
+    static constexpr uint8_t internal_bits{detail::sizeof_bits<decltype(data_)>};
+    //!\brief The number of bits reserved for representing size information in data_. 6 for `uint64_t`.
+    static constexpr uint8_t size_bits{detail::most_significant_bit_set(detail::next_power_of_two(internal_bits))};
```

Comment:
If you want to encode the `size` and `value` into one integer, I think it would be better to use [bitfields](https://en.cppreference.com/w/cpp/language/bit_field), because the compiler will do all the hard work for you and it is a standard feature!

https://godbolt.org/z/aNcMKl

Reply:
I didn't know about it, but I think it's preferable.